src/dolphin/atmosphere/troposphere.py

```diff
@@ -0,0 +1,570 @@
+    strides : Dict[str, int], optional
+        Strides for resampling, by default {"x": 1, "y": 1}.
+
+    Returns
+    -------
+    Dict[str, Path]
+        Dictionary of prepared geometry files.
+    """
+    geometry_dir.mkdir(exist_ok=True)
+
+    stitched_geo_list = {}
+
+    # local_incidence_angle needed by anyone?
+    datasets = ["los_east", "los_north"]
+
+    for ds_name in datasets:
+        outfile = geometry_dir / f"{ds_name}.tif"
+        logger.info(f"Creating {outfile}")
+        stitched_geo_list[ds_name] = outfile
+        ds_path = f"/data/{ds_name}"
+        cur_files = [io.format_nc_filename(f, ds_name=ds_path) for f in geo_files]
+
+        no_data = 0
+
+        stitching.merge_images(
+            cur_files,
+            outfile=outfile,
+            driver="GTiff",
+            out_bounds=out_bounds,
+            out_bounds_epsg=epsg,
+            in_nodata=no_data,
+            out_nodata=no_data,
+            target_aligned_pixels=True,
+            strides=strides,
+            resample_alg="nearest",
+            overwrite=False,
+        )
+
+    if dem_file:
+        height_file = geometry_dir / "height.tif"
+        stitched_geo_list["height"] = height_file
+        if not height_file.exists():
+            logger.info(f"Creating {height_file}")
+            stitching.warp_to_match(
+                input_file=dem_file,
+                match_file=matching_file,
+                output_file=height_file,
+                resample_alg="cubic",
+            )
+
+    return stitched_geo_list
+
+
+def compute_pyaps(delay_parameters: DelayParams) -> np.ndarray:
+    """Compute tropospheric delay datacube using PyAPS.
+
+    Parameters
+    ----------
+    delay_parameters : DelayParams
+        delay parameters and grid information.
+
+    Returns
+    -------
+    np.ndarray
+       tropospheric delay datacube.
+    """
+    import pyaps3 as pa
+
+    # X and y for the entire datacube
+    y_2d, x_2d = np.meshgrid(
+        delay_parameters.y_coordinates, delay_parameters.x_coordinates, indexing="ij"
+    )
+
+    # Lat/lon coordinates
+    lat_datacube, lon_datacube = oput.transform_xy_to_latlon(
+        delay_parameters.epsg, x_2d, y_2d
+    )
+
+    tropo_delay_datacube_list = []
+
+    for hgt in delay_parameters.z_coordinates:
+        dem_datacube = np.full(lat_datacube.shape, hgt)
+
+        # Delay for the reference image
+        ref_aps_estimator = pa.PyAPS(
+            fspath(delay_parameters.reference_file[0]),
+            dem=dem_datacube,
+            inc=0.0,
+            lat=lat_datacube,
+            lon=lon_datacube,
+            grib=delay_parameters.tropo_model,
+            humidity="Q",
+            model=delay_parameters.tropo_model,
+            verb=False,
+            Del=delay_parameters.delay_type,
+        )
+
+        phs_ref = np.zeros(
+            (ref_aps_estimator.ny, ref_aps_estimator.nx), dtype=np.float32
+        )
+        ref_aps_estimator.getdelay(phs_ref)
```

Comment:
is it possible to make this
```python
phase_reference = ref_aps_estimator.getdelay()
```
? 

I know this is copying the mintpy template, but it's a bit of a strange pattern to make an output array and then pass that as the input to modify in place (unless you're specifically trying to [save space like the numpy `out` argument](https://stackoverflow.com/questions/27293830/utility-of-parameter-out-in-numpy-functions)). here we're not trying to save/reuse space so we might as well let the function create the output

Reply:
you are right, yes it is possible